lib/chef/provider/windows_task.rb

```diff
@@ -51,9 +51,11 @@ def set_current_resource(task_hash)
         @current_resource.months(task_hash[:months]) if task_hash[:months]
         set_current_idle_time(task_hash[:idle_time]) if task_hash[:idle_time]
         @current_resource.random_delay(task_hash[:random_delay]) if task_hash[:random_delay]
-        @current_resource.execution_time_limit(task_hash[:execution_time_limit]) if task_hash[:execution_time_limit]
+        @current_resource.execution_time_limit(task_hash[:execution_time_limit] || "PT72H")
         @current_resource.status = :running if task_hash[:Status] == "Running"
         @current_resource.enabled = true if task_hash[:ScheduledTaskState] == "Enabled"
+        @current_resource.start_time = task_hash[:StartTime] if task_hash[:StartTime]
+        @current_resource.start_day = task_hash[:StartDate] if task_hash[:StartDate]
```

Comment:
`current_resource` set as an accessor in Chef::Provider, so these lines could be updated while you're here to not use the instance variable.

Reply:
@btm After replacing `@current_resource` as `current_resource` throws following error
```
>>>> Caused by NoMethodError: undefined method `exists=' for nil:NilClass
C:/Users/azure/chef/chef/lib/chef/provider/windows_task.rb:44:in `set_current_resource'
```
https://github.com/chef/chef/blob/master/lib/chef/provider/windows_task.rb#L44

Reply:
Change only line 33 to this:
```
self.current_resource = Chef::Resource::WindowsTask.new(new_resource.name)
```

I know this is confusing. What is happening here without the `self` is you're assigning `Chef::Resource::WindowsTask.new(new_resource.name)` to a local variable named `current_resource`. Once you're outside of `load_current_resource`, that variable no longer exists.

In this case, we already have an accessor method named `current_resource`, so it seems unnecessarily confusing for Ruby to also allow the creation of a local variable named `current_resource`, but it does. So we have be explicit that we mean the accessor method. Later when we're calling methods against the object stored in `current_resource`, we don't have to be explicit.

In most of our current code we leave the @ on this one line, which also works. Using the accessor instead of the instance variable makes it easier to change behavior later. We could change the accessor to a normal method and manipulate the data that we return. In this case, we'd end up going back and changing the one line of `@current_resource...` to `self.current_resource`.